src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java

```diff
@@ -0,0 +1,636 @@
+    }
+
+    private JButton getCloseButton() {
+        closeButton = new JButton(MessageBundle.getMessage("angal.common.close.btn"));
+        closeButton.setMnemonic(MessageBundle.getMnemonic("angal.common.close.btn.key"));
+        closeButton.addActionListener(actionEvent -> dispose());
+        return closeButton;
+    }
+
+    private JScrollPane getScrollPaneInventory() {
+        if (scrollPaneInventory == null) {
+            scrollPaneInventory = new JScrollPane();
+            scrollPaneInventory.setViewportView(getJTableInventory());
+        }
+        return scrollPaneInventory;
+    }
+
+    private JTable getJTableInventory() {
+        if (jTableInventory == null) {
+            jTableInventory = new JTable();
+            jTableInventory.setFillsViewportHeight(true);
+            jTableInventory.setModel(new InventoryBrowsingModel());
+			jTableInventory.getSelectionModel().addListSelectionListener(e -> {
+				if (e.getValueIsAdjusting()) {
+					int[] selectedRows = jTableInventory.getSelectedRows();
+					if (selectedRows.length == 1) {
+						editButton.setEnabled(true);
+						viewButton.setEnabled(true);
+						deleteButton.setEnabled(true);
+					} else {
+						editButton.setEnabled(false);
+						viewButton.setEnabled(false);
+						deleteButton.setEnabled(false);
+					}
+				}
+		    });
+        }
+        return jTableInventory;
+    }
+
+    class InventoryBrowsingModel extends DefaultTableModel {
+
+        /**
+         * 
+         */
+        private static final long serialVersionUID = 1L;
+
+        public InventoryBrowsingModel() {
+	        inventoryList = new ArrayList<>();
+	        String status = statusComboBox.getSelectedIndex() > 0 ? statusComboBox.getSelectedItem().toString().toLowerCase() : null;
+	        String type = InventoryType.ward.toString();
+	        try {
+		        inventoryList = medicalInventoryManager.getMedicalInventoryByParams(dateFrom, dateTo, status, type);
+	        } catch (OHServiceException e) {
+		        OHServiceExceptionUtil.showMessages(e);
+	        }
+        }
+
+		public InventoryBrowsingModel(int startIndex, int pageSize) {
+			inventoryList = new ArrayList<>();
+			String status = statusComboBox.getSelectedIndex() > 0 ? statusComboBox.getSelectedItem().toString().toLowerCase() : null;
+			String type = InventoryType.ward.toString();
+			try {
+				Page<MedicalInventory> medInventorypage = medicalInventoryManager.getMedicalInventoryByParamsPageable(dateFrom, dateTo, status, type,
+								startIndex, pageSize);
+				inventoryList = medInventorypage.getContent();
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+		}
+
+		@Override
+		public Class<?> getColumnClass(int c) {
+			if (c == 0) {
+				return String.class;
+			} else if (c == 1) {
+				return String.class;
+			} else if (c == 2) {
+				return String.class;
+			} else if (c == 3) {
+				return String.class;
+			} else if (c == 4) {
+				return String.class;
+			}
```

Comment:
wouldn't it be more efficient to test for the range 0 to 4 and return `String.class`, and else return `null`?

Reply:
just old fashion code, fixed.